### ISSUE TITLE
script error

### ISSUE THREAD

**Issue by blankered** [UAVenture/ros-setups#1]
Thank you for creating this script! But like every installation of ROS I perform on my Edison or Jetsen TK1, I always run into errors. I cloned the repository to my home directory and ran the ros-install script from the downloaded directory. I created the px4 user account as the directions said. Here is the error output I am receiving. 

........
Unpacking liblz4-1:i386 (from .../liblz4-1_0.0~r114-2~bpo70+1_i386.deb) ...
Selecting previously unselected package liblz4-dev:i386.
Unpacking liblz4-dev:i386 (from .../liblz4-dev_0.0~r114-2~bpo70+1_i386.deb) ...
Setting up liblz4-1:i386 (0.0~r114-2~bpo70+1) ...
Setting up liblz4-dev:i386 (0.0~r114-2~bpo70+1) ...
./install_ros.sh: line 56: cd: //ros_catkin_ws: No such file or directory

ERROR: your rosdep installation has not been initialized yet.  Please run:

```
rosdep update
```

“*****************************************************************_”
“About to start some heavy building. Go have a looong coffee break.”
“**_***************************************************************”
sudo: ./src/catkin/bin/catkin_make_isolated: command not found
./install_ros.sh: line 66: //.profile: Permission denied
./install_ros.sh: line 67: //.profile: No such file or directory
./install_ros.sh: line 69: //.bashrc: Permission denied
./install_ros.sh: line 70: //.bashrc: No such file or directory
./install_ros.sh: line 72: cd: //ros_catkin_ws: No such file or directory

Are there some packages I need to fix manually?

**Comment by SimonWilks** [comment]
Line 56 should read `cd ~/ros_catkin_ws` and not `cd //ros_catkin_ws`. Can you check if the `/home/px4/ros_catkin_ws` directory exists?

I also noticed the README said to run the script with sudo though it should catch this. I just updated the instructions.

If you don't won't any (or less) problems you might want to just copy paste each line one at a time.

**Comment by blankered** [comment]
I think the first issue is that I have no "px4" user directory. When I login using the created px4 account information, it redirects me the /home directory. Would this cause issues? I will reflash the board and attempt to follow the instructions again.

Debian GNU/Linux 7 ubilinux ttyMFD2

ubilinux login: px4
Password:
Last login: Tue Mar  3 19:28:37 UTC 2015 on ttyMFD2
Linux ubilinux 3.10.17-poky-edison-ww42+ #4 SMP PREEMPT Wed Oct 29 12:41:25 GMT 2014 i686

The programs included with the Debian GNU/Linux system are free software;
the exact distribution terms for each program are described in the
individual files in /usr/share/doc/*/copyright.

Debian GNU/Linux comes with ABSOLUTELY NO WARRANTY, to the extent
permitted by applicable law.
No directory, logging in with HOME=/
.

**Comment by SimonWilks** [comment]
Ah! I see the problem. In the instructions I say `useradd` but unlike the `adduser` app based on it it will not create a home directory. Will update the instructions again in a sec.

If you don't mind reflashing that would be great. My board is operational at the moment and I can't retest at the moment.

**Comment by SimonWilks** [comment]
Done. And yes it will have messed everything up.

**Comment by blankered** [comment]
Im running through the instructions again. One thing I think might be useful to add is:
1. How to mount microsd card "
2. How to change the /var/apt/cache directory to the mounted sd card 
3. How to change the symbolic directories to forward to the mounted card
while this feature would free up space on the board

I think the instructions are as follows, maybe wrong
1.) fdisk -l                 look for sd card id
2.) mount -t vfat /dev/(sd mount id)      /mnt
3.) copy directory /var/cache/apt  to /mnt
4.) ln -s /mnt/apt/archieves /var/cache/apt/archieves

**Comment by SimonWilks** [comment]
I'm using the breakout board which doesn't have an SD card slot. Happy to add the instructions once you verify them.

**Comment by SimonWilks** [comment]
@blankered any luck with the installation? If not I just received a second Edison that I can use for testing and I can run through it and clean it up tomorrow.

**Comment by blankered** [comment]
The script works perfectly! Nice work. Now I am trying to figure out how to free up some space now. I was exploring options about booting from the SD card directly but I have had little success on finding resources about doing this.

**Comment by SimonWilks** [comment]
Great. Thanks for testing it :)

One thing you can move is the cache directory which may give a fair bit of space back:

```
mv /var/cache /home
cd /var
ln -sf /home/cache cache
```

**Comment by blankered** [comment]
I am having problems running the "roslaunch mavros px4.launch" command from the "ros_catkin_ws" folder. I moved my /opt/ros/indigo to /mnt/indigo with the following similar commands above to make more space. I simply get command not found with I run in normal and sudo mode. I am not sure how to fix this issue. I will try and move the indigo folder back and report back.

EDIT: I didn't properly forward the directories. I fixed that "ln -sf /mnt/indigo /opt/ros/indigo" while in super user mode. The "roslaunch mavros px4.launch" command now works. And I have a bunch of space!
